evo_researcher/benchmark/utils.py

```diff
@@ -19,6 +20,39 @@ class PredictionResult(BaseModel):
     cost: t.Optional[float] = None
 
 
+AgentPredictions = t.Dict[str, PredictionResult]
+Predictions = t.Dict[str, AgentPredictions]
+
+
+class PredictionsCache(BaseModel):
+    predictions: Predictions
+
+    def save(self, path: str):
+        # If the file exists, load it and add the new predictions
+        if os.path.exists(path):
+            old_cache = self.parse_file(path)
+            for agent, agent_predictions in self.predictions.items():
+                for question, prediction in agent_predictions.items():
+                    old_cache.predictions[agent][question] = prediction
+            self = old_cache
```

Comment:
Is this doing something? I'd expect `self = old_cache` will only change the variable within this scope, not update the model itself 🤔

Reply:
Yes, but we use the updated `self` in this scope just below with `self.json()`

Reply:
Ah, I see! That's a little self-confusing 😄